{
  try {
    daemon.join();
  }
 catch (  InterruptedException e) {
  }
}
